### PR TITLE
Remove Python27 support and fix for arrival/departure as datetime objects

### PR DESCRIPTION
**Pull request by eifinger**
If arrival or departure are provided as datetime.datetime objects then urllib will encode it in a format that HERE does not accept.

I implemented a fix which forces the conversion to an ISO8601 string.

Took me 2h to figure this out 😂

### CODE REVIEW

File: herepy/routing_api.py [hunk 1]
```diff
@@ -309,6 +312,13 @@ def matrix(self,
         response = self.__get(self.URL_CALCULATE_MATRIX, data, RoutingMatrixResponse)
         return response
 
+    @staticmethod
+    def _convert_datetime_to_isoformat(datetime_object):
+      """Convert a datetime.datetime object to an ISO8601 string."""
+      if isinstance(datetime_object, datetime.datetime):
+        datetime_object = datetime_object.isoformat()
+      return datetime_object
```
Comment (abdullahselek) [inline herepy/routing_api.py]: Please leave 4 empty spaces not 2 :)

File: herepy/routing_api.py [hunk 2]
```diff
@@ -309,6 +312,13 @@ def matrix(self,
         response = self.__get(self.URL_CALCULATE_MATRIX, data, RoutingMatrixResponse)
         return response
 
+    @staticmethod
+    def _convert_datetime_to_isoformat(datetime_object):
+      """Convert a datetime.datetime object to an ISO8601 string."""
+      if isinstance(datetime_object, datetime.datetime):
```
Comment (abdullahselek) [inline herepy/routing_api.py]: Leave an empty line after description.

File: herepy/routing_api.py [hunk 3]
```diff
@@ -309,6 +312,13 @@ def matrix(self,
         response = self.__get(self.URL_CALCULATE_MATRIX, data, RoutingMatrixResponse)
         return response
 
+    @staticmethod
+    def _convert_datetime_to_isoformat(datetime_object):
```
Comment (abdullahselek) [inline herepy/routing_api.py]: Please add some unit tests for this method covering date and str parameters.

**Comment by eifinger** [comment]
Python 2.7 is failing. I _could_ of course try to provide a fix. But Python 2.7 end of life in 21 days^^

**Comment by abdullahselek** [comment]
Yeah we can remove Python 2.7 from CI configurations.

**Comment by eifinger** [comment]
I will remove it from the CI files and push another commit.
Should we also remove all the python 27 things like `from __future__`?

**Comment by eifinger** [comment]
I removed all remnants of Python 2.7 support I could find.
I also removed the encoding information according to this answer:
https://stackoverflow.com/questions/14083111/should-i-use-encoding-declaration-in-python-3

**Comment by eifinger** [comment]
If this is okay for you I would recommend to release a new major version as Python27 support is dropped.

**Comment by abdullahselek** [comment]
Changes look good, yes I will release new major version.

**Pull request merged**